Api/Resource.php

```diff
@@ -50,6 +50,34 @@ class Resource implements ResourceInterface
      */
     private $validationGroups;
     /**
+     * @var bool
+     */
+    private $paginationEnabledByDefault = false;
+    /**
+     * @var bool
+     */
+    private $clientAllowedToEnablePagination = false;
+    /**
+     * @var float
+     */
+    private $itemsPerPageByDefault = 30.;
```

Comment:
I agree with @felicitus as it's also a broken modelling since we can't have half of items per page :)